internal/provider/client.go

```diff
@@ -18,22 +19,36 @@ func NewSquaredUpClient(region string, apiKey string) (*SquaredUpClient, error)
 		return nil, err
 	}
 
-	return &SquaredUpClient{
+	req, err := http.NewRequest("GET", baseURL+"/api/plugins/latest", nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create request: %v", err)
+	}
+
+	client := &http.Client{}
+
+	squaredUpClient := &SquaredUpClient{
 		baseURL:    baseURL,
 		apiKey:     apiKey,
-		httpClient: &http.Client{},
-	}, nil
+		httpClient: client,
+	}
+
+	_, err = squaredUpClient.doRequest(req)
+	if err != nil {
+		return nil, fmt.Errorf("invalid api key with the provided region. check the api key and region and try again")
+	}
+
+	return squaredUpClient, nil
 }
 
 func determineBaseURL(region string) (string, error) {
-	switch region {
-	case "us":
+	if region == "us" {
 		return "https://api.squaredup.com", nil
-	case "eu":
+	} else if region == "eu" {
 		return "https://eu.api.squaredup.com", nil
-	default:
-		return "", fmt.Errorf("unsupported region: %s", region)
+	} else if strings.HasPrefix(region, "https://") {
+		return region, nil
```

Comment:
I don't know much about GO, but if its easy, if might be worth adding a check that there is no trailing slash, or removing it.  Otherwise, that would be an easy thing to note in our internal documentation.

Reply:
Good idea, I think we should remove the trailing slash. I have add that to commit [2b85c61](https://github.com/squaredup/terraform-provider-squaredup/pull/42/commits/2b85c6152ed7975fac375e5dd022a0fff35a16c5)